static/app/components/events/interfaces/analyzeFrames.tsx

```diff
@@ -0,0 +1,205 @@
+    ],
+    resources: tct(
+      'Optimize cold/warm app starts by offloading operations off the main thread and [link:lazily initializing] components.',
+      {
+        link: (
+          <ExternalLink href="https://developer.android.com/topic/performance/vitals/launch-time#heavy-app" />
+        ),
+      }
+    ),
+  },
+  {
+    module: 'android.content.res.AssetManager',
+    functions: [
+      'nativeOpenAsset',
+      'nativeOpenAssetFd',
+      'nativeOpenNonAsset',
+      'nativeOpenNonAssetFd',
+    ],
+    offendingThreadStates: [
+      ThreadStates.WAITING,
+      ThreadStates.TIMED_WAITING,
+      ThreadStates.RUNNABLE,
+    ],
+    resources: t('If possible, move loading heavy assets off the main thread.'),
+  },
+  {
+    module: 'android.content.res.AssetManager',
+    functions: [/^nativeGetResource[a-zA-Z]+/],
+    offendingThreadStates: [
+      ThreadStates.WAITING,
+      ThreadStates.TIMED_WAITING,
+      ThreadStates.RUNNABLE,
+    ],
+    resources: t('Look for heavy resources in the /res or /res/raw folders.'),
+  },
+  {
+    module: 'android.view.LayoutInflater.inflate',
+    offendingThreadStates: [
+      ThreadStates.WAITING,
+      ThreadStates.TIMED_WAITING,
+      ThreadStates.RUNNABLE,
+    ],
+    resources: tct(
+      'The app is potentially inflating a heavy, deeply-nested layout. [link:Optimize view hierarchy], use view stubs, use include/merge tags for reusing inflated views.',
+      {
+        link: (
+          <ExternalLink href="https://developer.android.com/develop/ui/views/layout/improving-layouts" />
+        ),
+      }
+    ),
+  },
+];
+
+function satisfiesModuleCondition(frame: Frame, suspect: SuspectFrame) {
+  if (isNil(suspect.module)) {
+    return true;
+  }
+  const matchFuction = suspect.module;
+  return typeof matchFuction === 'string'
+    ? frame.module?.startsWith(matchFuction)
+    : frame.module && matchFuction.test(frame.module);
+}
+
+function satisfiesFunctionCondition(frame: Frame, suspect: SuspectFrame) {
+  if (isNil(suspect.functions) || suspect.functions.length === 0) {
+    return true;
+  }
+  if (isNil(frame.function)) {
+    return false;
+  }
+  for (let index = 0; index < suspect.functions.length; index++) {
+    const matchFuction = suspect.functions[index];
+    const match =
+      typeof matchFuction === 'string'
+        ? frame.function === matchFuction
+        : matchFuction.test(frame.function);
+    if (match) {
+      return true;
+    }
+  }
+  return false;
+}
+
+function satisfiesOffendingThreadCondition(
+  threadState: string | undefined | null,
+  suspect: SuspectFrame
+) {
+  if (
+    isNil(suspect.offendingThreadStates) ||
+    suspect.offendingThreadStates.length === 0
+  ) {
+    return true;
+  }
+  const mappedState = getMappedThreadState(threadState);
+
+  if (isNil(mappedState)) {
+    return false;
+  }
+  return suspect.offendingThreadStates.includes(mappedState);
+}
+
+export function analyzeFramesForRootCause(event: Event): {
```

Comment:
Few more conditions I need to add from [this](https://www.notion.so/sentry/ANR-root-cause-analysis-f8e9352861ce4e039a58b0a1952285de), will follow up